lib/assets/data_frame_cell/main.js

```diff
@@ -0,0 +1,563 @@
+              stroke="currentColor"
+              xmlns="http://www.w3.org/2000/svg"
+              viewBox="0 0 16 10"
+              aria-hidden="true"
+            >
+              <path
+                d="M15 1.2l-7 7-7-7"
+              />
+            </svg>
+          </button>
+          <button
+            class="button button--sm"
+            @click="$emit('removeOperation')"
+            type="button"
+          >
+            <svg
+              class="button-svg"
+              fill="currentColor"
+              stroke="none"
+              xmlns="http://www.w3.org/2000/svg"
+              viewBox="0 0 16 16"
+              aria-hidden="true"
+            >
+              <path
+                d="M11.75 3.5H15.5V5H14V14.75C14 14.9489 13.921 15.1397 13.7803 15.2803C13.6397 15.421 13.4489
+                15.5 13.25 15.5H2.75C2.55109 15.5 2.36032 15.421 2.21967 15.2803C2.07902 15.1397 2 14.9489 2
+                14.75V5H0.5V3.5H4.25V1.25C4.25 1.05109 4.32902 0.860322 4.46967 0.71967C4.61032 0.579018 4.80109
+                0.5 5 0.5H11C11.1989 0.5 11.3897 0.579018 11.5303 0.71967C11.671 0.860322 11.75 1.05109 11.75
+                1.25V3.5ZM12.5 5H3.5V14H12.5V5ZM5.75 7.25H7.25V11.75H5.75V7.25ZM8.75
+                7.25H10.25V11.75H8.75V7.25ZM5.75 2V3.5H10.25V2H5.75Z"
+              />
+            </svg>
+          </button>
+        </span>
+      </div>
+    </div>
+  `,
+  };
+
+  const BaseButton = {
+    name: "BaseButton",
+    props: {
+      label: {
+        type: String,
+        default: "",
+      },
+    },
+    template: `
+    <button class="button button--sm button--dashed" type="button" :disabled="noDataFrame"
+      @click="$emit('addOperation')">
+      <svg width="10" height="10" viewBox="0 0 10 10" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
+        <path d="M4.41699 4.41602V0.916016H5.58366V4.41602H9.08366V5.58268H5.58366V9.08268H4.41699V5.58268H0.916992V4.41602H4.41699Z"/>
+      </svg>
+      {{ label }}
+    </button>
+    `,
+  };
+
+  const BaseInputIcon = {
+    name: "BaseInputIcon",
+    props: {
+      label: {
+        type: String,
+        default: "",
+      },
+    },
+    template: `
+    <div class="icon-container" @click="$emit('removeOperation')">
+      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="16" height="16">
+        <path fill="none" d="M0 0h24v24H0z"/>
+        <path d="M17 6h5v2h-2v13a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V8H2V6h5V3a1 1 0 0 1 1-1h8a1 1 0 0 1 1 1v3zm1 2H6v12h12V8zm-9 3h2v6H9v-6zm4 0h2v6h-2v-6zM9 4v2h6V4H9z"/>
+      </svg>
+    </div>
+    `,
+  };
+
+  const app = Vue.createApp({
+    components: { BaseSelect, BaseInput, Accordion, BaseButton, BaseInputIcon },
+    template: `
+      <div class="app">
+        <div class="box box-warning" v-if="missingDep">
+          <p>
+            To successfully work with data frames, you need to add the following dependency:
+          </p>
+          <pre><code>{{ missingDep }}</code></pre>
```

Comment:
Do we really need this? kino_explorer depends on explorer, so it should always be there.

Reply:
Oh! That’s true 🙈